iroh-bytes/src/protocol/range_spec.rs

```diff
@@ -167,7 +200,11 @@ impl RangeSpecSeq {
         Self(res)
     }
 
-    /// An infinite iterator of range specs
+    /// An infinite iterator of range specs for blobs in the collection.
+    ///
+    /// Each item yielded by the iterator is the [`RangeSpec`] for a blob in the collection.
```

Comment:
I would leave out the notion of collections entirely. It is really simplest to think of this as just a fancy/efficient way to write down a (possibly infinite) sequence of RangeSets.

The way this is mapped to the notion of collections is that the first element is the collection itself, and all subsequent elements are children of the collection. But e.g. if we ever have collections in collections you might have a different scheme.

Reply:
no more collections mentioned